[Safe CPP] Re-land Address Warnings in RenderTableCol.cpp

stwrt opened the pull request:
#### 4767bd89fe19b3ab7f1cad692110b244051bad0e
<pre>
[Safe CPP] Re-land Address Warnings in RenderTableCol.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=294030">https://bugs.webkit.org/show_bug.cgi?id=294030</a>
<a href="https://rdar.apple.com/problem/152583157">rdar://problem/152583157</a>

Reviewed by Chris Dumez.

Re-land Address Warnings in RenderTableCol.cpp.

* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebCore/rendering/RenderElement.h:
(WebCore::RenderElement::checkedStyle const):
* Source/WebCore/rendering/RenderTableCol.cpp:
(WebCore::RenderTableCol::styleDidChange):
(WebCore::RenderTableCol::updateFromElement):
(WebCore::RenderTableCol::insertedIntoTree):
(WebCore::RenderTableCol::willBeRemovedFromTree):
(WebCore::RenderTableCol::clippedOverflowRect const):
(WebCore::RenderTableCol::clearNeedsPreferredLogicalWidthsUpdate):
(WebCore::RenderTableCol::checkedTable const):
(WebCore::RenderTableCol::nextColumn const):
(WebCore::RenderTableCol::borderAdjoiningCellStartBorder const):
(WebCore::RenderTableCol::borderAdjoiningCellEndBorder const):
(WebCore::RenderTableCol::borderAdjoiningCellBefore const):
(WebCore::RenderTableCol::borderAdjoiningCellAfter const):
(WebCore::RenderTableCol::offsetLeft const):
(WebCore::RenderTableCol::offsetTop const):
(WebCore::RenderTableCol::offsetWidth const):
(WebCore::RenderTableCol::offsetHeight const):
* Source/WebCore/rendering/RenderTableCol.h:

Canonical link: <a href="https://commits.webkit.org/295850@main">https://commits.webkit.org/295850@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d36c8de9c84fc0e3ca1178ba89aaed9d109c5040

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106273 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26022 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16418 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111471 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/56869 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/108312 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/26685 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34525 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80726 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109277 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21116 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/95902 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61053 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20638 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/14001 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56309 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90468 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14036 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114331 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33411 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/24638 "Found 2 new test failures: platform/mac/fast/text/line-break-locale.html svg/zoom/page/text-with-non-scaling-stroke.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89798 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33775 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92131 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89499 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22836 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34374 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12186 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/28995 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33336 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/38748 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33082 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36435 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34680 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/rendering/RenderElement.h:
```diff
@@ -68,6 +69,7 @@ class RenderElement : public RenderObject {
     bool hasInitializedStyle() const { return m_hasInitializedStyle; }
 
     const RenderStyle& style() const { return m_style; }
+    const CheckedRef<const RenderStyle> checkedStyle() const { return m_style; }
```
@rniwa It doesn't seem like this should be needed? m_style is a plain data member, it stays alive as long as the RenderElement.

cdumez replied:
Ryosuke thinks this won't be required after https://github.com/llvm/llvm-project/pull/142485. This seems like a static analysis false positive.

cdumez replied:
Seems we likely want a FIXME comment.

stwrt committed:
[Safe CPP] Re-land Address Warnings in RenderTableCol.cpp
https://bugs.webkit.org/show_bug.cgi?id=294030
rdar://problem/152583157

Reviewed by Chris Dumez.

Re-land Address Warnings in RenderTableCol.cpp.

* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebCore/rendering/RenderElement.h:
(WebCore::RenderElement::checkedStyle const):
* Source/WebCore/rendering/RenderTableCol.cpp:
(WebCore::RenderTableCol::styleDidChange):
(WebCore::RenderTableCol::updateFromElement):
(WebCore::RenderTableCol::insertedIntoTree):
(WebCore::RenderTableCol::willBeRemovedFromTree):
(WebCore::RenderTableCol::clippedOverflowRect const):
(WebCore::RenderTableCol::clearNeedsPreferredLogicalWidthsUpdate):
(WebCore::RenderTableCol::checkedTable const):
(WebCore::RenderTableCol::nextColumn const):
(WebCore::RenderTableCol::borderAdjoiningCellStartBorder const):
(WebCore::RenderTableCol::borderAdjoiningCellEndBorder const):
(WebCore::RenderTableCol::borderAdjoiningCellBefore const):
(WebCore::RenderTableCol::borderAdjoiningCellAfter const):
(WebCore::RenderTableCol::offsetLeft const):
(WebCore::RenderTableCol::offsetTop const):
(WebCore::RenderTableCol::offsetWidth const):
(WebCore::RenderTableCol::offsetHeight const):
* Source/WebCore/rendering/RenderTableCol.h:

Canonical link: https://commits.webkit.org/295850@main
